gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy

```diff
@@ -258,18 +265,14 @@ class GateConfig extends RedisHttpSessionConfiguration {
     createClient("swabbie", SwabbieService, okHttpClient)
   }
 
-  private <T> T createClient(String serviceName,
-                             Class<T> type,
-                             OkHttpClient okHttpClient,
-                             String dynamicName = null,
-                             boolean forceEnabled = false) {
+  static Endpoint createEndpoint(ServiceConfiguration serviceConfiguration,
```

Comment:
I extracted this into a static method so it can be used in `NotificationService` to find the endpoint for `echo` when relaying notification callbacks using an `OkHttpClient` instead of an `EchoService` (for reasons explained in code comments there).